libvcx/src/api_lib/api_c/credential_def.rs

```diff
@@ -110,6 +113,102 @@ pub extern fn vcx_credentialdef_create(command_handle: CommandHandle,
     error::SUCCESS.code_num
 }
 
+#[no_mangle]
+pub extern fn vcx_credentialdef_generate_and_store(command_handle: CommandHandle,
+                                                   source_id: *const c_char,
+                                                   schema_id: *const c_char,
+                                                   issuer_did: *const c_char,
+                                                   tag: *const c_char,
+                                                   revocation_details: *const c_char,
+                                                   cb: Option<extern fn(xcommand_handle: CommandHandle, err: u32, credentialdef_handle: u32)>) -> u32 {
+    info!("vcx_credentialdef_generate_and_store >>>");
+
+    check_useful_c_callback!(cb, VcxErrorKind::InvalidOption);
+    check_useful_c_str!(source_id, VcxErrorKind::InvalidOption);
+    check_useful_c_str!(schema_id, VcxErrorKind::InvalidOption);
+    check_useful_c_str!(tag, VcxErrorKind::InvalidOption);
+    check_useful_c_str!(revocation_details, VcxErrorKind::InvalidOption);
+
+    let issuer_did: String = if !issuer_did.is_null() {
+        check_useful_c_str!(issuer_did, VcxErrorKind::InvalidOption);
+        issuer_did.to_owned()
+    } else {
+        match settings::get_config_value(settings::CONFIG_INSTITUTION_DID) {
```

Comment:
Let's not leak `settings::` into new functions in `libvcx`. I think this issuer_did resolution should be put on `aries_vcx` layeer